From 5179eabbbea77b98bab94145d0cf9d911e6eba7f Mon Sep 17 00:00:00 2001
From: Self-Hosting-Group <selfhostinggroup-git+openwrt@shost.ing>
Date: Mon, 29 Dec 2025 00:00:00 +0000
Subject: [PATCH 5/7] luci-app-upnp: add second CGNAT UCI option

Alternative option to STUN allow-filtered. As requested by AquanJSW, to
test with Tailscale. Also adds the required daemon fix. No STUN public
IPv4 address detection; issues with multiple clients, e.g. PCP/NAT-PMP

(proposed for inclusion, to merge with prior)

Signed-off-by: Self-Hosting-Group <selfhostinggroup-git+openwrt@shost.ing>
---
 .../luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js b/applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js
index 0b305c32c2..cb0a87e78a 100644
--- a/applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js
+++ b/applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js
@@ -174,6 +174,7 @@ return view.extend({
 		o.value('', _('Disabled'), _('Override public/external IPv4 address to prevent a private'));
 		o.value('1', _('Enabled'), _('Filtering test currently requires an extra firewall rule'));
 		o.value('allow-filtered', _('Enabled') + ' (' + _('allow filtered') + ')', _('Allow filtered IPv4 CGNAT test result'));
+		o.value('allow-private-ext-ipv4', _('Enabled') + ' (' + _('avoid, allow private external IPv4') + ')', _('No STUN public IPv4 address detection; issues with multiple clients'));
 		o.optional = true;
 
 		o = s.taboption('advanced', form.Value, 'stun_host', _('STUN server'));
-- 
2.52.0.windows.1

